sshd-core/src/main/java/org/apache/sshd/common/util/io/IoUtils.java

```diff
@@ -257,6 +260,56 @@ public static void setPermissionsToFile(File f, Collection<PosixFilePermission>
     }
 
     /**
+     * <P>Get file owner.</P>
+     *
+     * @param path  The {@link Path}
+     * @param options The {@link LinkOption}s to use when querying the owner
+     * @return Owner of the file or null if unsupported
+     * @throws IOException If failed to access the file system
+     */
+    public static String getFileOwner(Path path, LinkOption... options) throws IOException {
+        String owner = null;
+        try {
+            UserPrincipal principal = Files.getOwner(path, options);
+            if (principal != null) {
+                owner = principal.getName();
+            }
+        } catch (UnsupportedOperationException e) {
+            // ignore
+        }
+        return owner;
+    }
+
+    /**
+     * <P>Get current user by file ownership.</P>
+     *
+     * @return Owner by creating a file and check ownership
+     * @throws IOException If failed to access the file system
+     */
+    public static String getCurrentUser() throws IOException {
+        synchronized (CURRENT_USER_HOLDER) {
+            String owner = CURRENT_USER_HOLDER.get();
+            if (owner == null) {
+                File t = null;
+                try {
+                    t = File.createTempFile("ssh-", ".tmp");
+                    owner = getFileOwner(t.toPath());
+                    if (owner == null) {
+                        // mark as attempted
+                        owner = "";
+                    }
+                } finally {
+                    if (t != null) {
+                        t.delete();
+                    }
+                }
+                CURRENT_USER_HOLDER.set(owner);
+            }
```

Comment:
delete does not throw IOException.

and I considered using user.name but then there is an issue with mismatch between java way of resolving user via environment and such and how resolve back id may look like.

so I decided to play safe.

if you think that user.name is sufficient I will use it, but I feel like it is insufficient.